diff --git a/node_modules/@atproto/api/dist/agent.js b/node_modules/@atproto/api/dist/agent.js
index 634e463..57b5c74 100644
--- a/node_modules/@atproto/api/dist/agent.js
+++ b/node_modules/@atproto/api/dist/agent.js
@@ -593,7 +593,7 @@ class Agent extends xrpc_1.XrpcClient {
                 hideAllFeeds: false,
             },
         };
-        const res = await this.app.bsky.actor.getPreferences({});
+        const res = await this.app.bsky.actor.getPreferences({}, {headers: {'atproto-proxy': ''}});
         const labelPrefs = [];
         for (const pref of res.data.preferences) {
             if (predicate.isValidAdultContentPref(pref)) {
@@ -1275,14 +1275,14 @@ class Agent extends xrpc_1.XrpcClient {
     async updatePreferences(cb) {
         try {
             await __classPrivateFieldGet(this, _Agent_prefsLock, "f").acquireAsync();
-            const res = await this.app.bsky.actor.getPreferences({});
+            const res = await this.app.bsky.actor.getPreferences({}, {headers: {'atproto-proxy': ''}});
             const newPrefs = cb(res.data.preferences);
             if (newPrefs === false) {
                 return res.data.preferences;
             }
             await this.app.bsky.actor.putPreferences({
                 preferences: newPrefs,
-            });
+            }, {headers: {'atproto-proxy': ''}});
             return newPrefs;
         }
         finally {
