diff --git a/node_modules/react-native-reanimated/Common/cpp/worklets/SharedItems/Shareables.cpp b/node_modules/react-native-reanimated/Common/cpp/worklets/SharedItems/Shareables.cpp
index f8c5bfc..47b5fbe 100644
--- a/node_modules/react-native-reanimated/Common/cpp/worklets/SharedItems/Shareables.cpp
+++ b/node_modules/react-native-reanimated/Common/cpp/worklets/SharedItems/Shareables.cpp
@@ -1,5 +1,7 @@
 #include <worklets/SharedItems/Shareables.h>
 
+#include <algorithm>
+
 using namespace facebook;
 
 namespace worklets {
